File: code/functions/rprism/tests/testthat/test-load_rna.R
test_that("load_rna_gex() met500 works", {
  skip_on_cran()

  cat("\n")
  df_rna <- load_rna_gex(study="met500", level="genes", metric="TPM", test_mode=TRUE)
  expect_true(is(df_rna, "data.frame"))

  cat("\n")
  df_rna <- load_rna_gex(study="met500",  level="genes", metric="counts", test_mode=TRUE,
                         identifiers=c("MO_1031", "MO_1107"), identifiers_name="Subject_Id")
  expect_true(is(df_rna, "data.frame"))

})


test_that("load_rna_fus() met500 works", {
  skip_on_cran()

  cat("\n")
  df_rna <- load_rna_fus(study="met500", mode="arriba")
  expect_true(is(df_rna, "data.frame"))

  cat("\n")
  df_rna <- load_rna_fus(study="met500", mode="pizzly")
  expect_true(is(df_rna, "data.frame"))
})


test_that("load_rna_gex() prism works", {
  skip_on_cran()

  cat("\n")
  df_rna <- load_rna_gex(study="prism",  level="genes", metric="TPM", test_mode=TRUE)
  expect_true(is(df_rna, "data.frame"))

  cat("\n")
  df_rna <- load_rna_gex(study="prism",  level="genes", metric="counts", test_mode=TRUE,
                         identifiers=c("201410666GH", "200905404EN"), identifiers_name="Subject_Id")
  expect_true(is(df_rna, "data.frame"))
})


test_that("load_rna_fus() prism works", {
  skip_on_cran()

  cat("\n")
  df_rna <- load_rna_fus(study="prism", mode="ericscript")
  expect_true(is(df_rna, "data.frame"))

  cat("\n")
  df_rna <- load_rna_fus(study="prism", mode="starfusion")
  expect_true(is(df_rna, "data.frame"))
})


test_that("load_rna_gex() tcga works", {
  skip_on_cran()

  cat("\n")
  df_rna <- load_rna_gex(study="tcga",  metric="TPM", test_mode=TRUE)
  expect_true(is(df_rna, "data.frame"))

  cat("\n")
  df_rna <- load_rna_gex(study="tcga",  metric="counts", test_mode=TRUE)
  expect_true(is(df_rna, "data.frame"))

  cat("\n")
  df_rna <- load_rna_gex(study="tcga",  metric="TPM", test_mode=TRUE,
                     identifiers=c("TCGA-02-0047-01A-01R-1849-01", "TCGA-3X-AAV9-01A-72R-A41I-07"),
                     identifiers_name="Sample_Id")
  expect_true(is(df_rna, "data.frame"))
  expect_equal(ncol(df_rna), 3)
})


test_that("load_rna_fus() tcga works", {
  skip_on_cran()

  cat("\n")
  df_rna <- load_rna_fus(study="tcga", mode="aggregated")
  expect_true(is(df_rna, "data.frame"))
})


test_that("load_rna_fus() tcga_6_samples works", {
  skip_on_cran()

  cat("\n")
  df_rna <- load_rna_fus(study="tcga_6_samples", mode="fusioncatcher")
  expect_true(is(df_rna, "data.frame"))
})

File: code/functions/pyprism/pyprism/data/tests/test_load_rna.py
# -*- coding: utf-8 -*-
"""
Created on Wed Sep 30 2020

@author: Yoann Pradat

Tests for _load_rna.py module.
"""

from pyprism.data import load_rna_fus, load_rna_gex

def test_load_rna_gex_met500():
    df_rna = load_rna_gex(study="met500", metric="counts", test_mode=True)
    del df_rna

    df_rna = load_rna_gex(study="met500", metric="TPM", test_mode=True, identifiers=["MO_1031", "MO_1107"],
                      identifiers_name="Subject_Id")
    del df_rna


def test_load_rna_fus_met500():
    df_rna = load_rna_fus(study="met500", mode="arriba")
    df_rna = load_rna_fus(study="met500", mode="pizzly")
    df_rna = load_rna_fus(study="met500", mode="oncokb_civic")


def test_load_rna_gex_prism():
    df_rna = load_rna_gex(study="prism", metric="TPM", test_mode=True)
    del df_rna

    df_rna = load_rna_gex(study="prism", metric="TPM", test_mode=True, identifiers=["201410666GH", "200905404EN"],
                      identifiers_name="Subject_Id")
    del df_rna


def test_load_rna_fus_prism():
    df_rna = load_rna_fus(study="prism", mode="ericscript")
    df_rna = load_rna_fus(study="prism", mode="starfusion")
    df_rna = load_rna_fus(study="prism", mode="oncokb_civic")


def test_load_rna_gex_tcga():
    df_rna = load_rna_gex(study="tcga", metric="TPM", test_mode=True)
    del df_rna

    df_rna = load_rna_gex(study="tcga", metric="TPM", test_mode=True,
                      identifiers=["TCGA-02-0047-01A-01R-1849-01", "TCGA-3X-AAV9-01A-72R-A41I-07"],
                      identifiers_name="Sample_Id")
    del df_rna


def test_load_rna_fus_tcga():
    df_rna = load_rna_fus(study="tcga", mode="aggregated")


def test_load_rna_gex_tcga_6_samples():
    df_rna = load_rna_gex(study="tcga_6_samples", metric="TPM", other_mode="prism")
    del df_rna

    df_rna = load_rna_gex(study="tcga_6_samples", metric="counts", other_mode="zheng")
    del df_rna
